lib/OpenQA/Worker/Engines/isotovideo.pm

```diff
@@ -345,33 +347,48 @@ sub engine_workit {
         my $error = locate_local_assets(\%vars, $assetkeys, $pooldir);
         return $error if $error;
     }
-    my $casedir     = testcasedir($vars{DISTRI}, $vars{VERSION}, $shared_cache);
-    my $productdir  = productdir($vars{DISTRI}, $vars{VERSION}, $shared_cache);
-    my $target_name = path($casedir)->basename;
-
-    $vars{ASSETDIR} //= OpenQA::Utils::assetdir();
-    $vars{CASEDIR}  //= $vars{ABSOLUTE_TEST_CONFIG_PATHS} ? $casedir : $target_name;
+    $vars{ASSETDIR} //= OpenQA::Utils::assetdir;
+
+    # ensure a CASEDIR and a PRODUCTDIR is assigned and create a symlink if required
+    my $absolute_paths        = $vars{ABSOLUTE_TEST_CONFIG_PATHS};
+    my @vars_for_default_dirs = ($vars{DISTRI}, $vars{VERSION}, $shared_cache);
+    my $default_casedir       = testcasedir(@vars_for_default_dirs);
+    my $default_productdir    = productdir(@vars_for_default_dirs);
+    my $target_name           = path($default_casedir)->basename;
+    my $has_custom_dir        = $vars{CASEDIR} || $vars{PRODUCTDIR};
+    my $casedir               = $vars{CASEDIR} //= $absolute_paths ? $default_casedir : $target_name;
+    if ($casedir eq $target_name) {
+        $vars{PRODUCTDIR} //= substr($default_productdir, rindex($default_casedir, $target_name));
+        if (my $error = _link_repo($default_casedir, $pooldir, $target_name)) { return $error }
+    }
+    else {
+        $vars{PRODUCTDIR} //= $absolute_paths ? $default_productdir : abs2rel($default_productdir, $default_casedir);
+    }
```

Comment:
Instead of this if/else here I would prefer a function with a good name and maybe exception handling instead of error value checking and returning. Could you please give it a try? If it doesn't work keep it as is

Reply:
The whole if/else-block is *not* about error handling but about distinguishing different cases (none of the cases is an error case). Changing the error handling of `_link_repo` to use exceptions to get rid of the nested if makes not much sense because then we needed a try-catch here unless we change the error handling of the overall function. But that's out of scope for this PR as I don't want to change too much at a time. And even if that wasn't the case, keep in mind that we return a hashref in the error case which is handled by the outer function specifically so I'd say there's not much to gain.

Reply:
I mainly meant to follow https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29#comments-rules or https://blog.usejournal.com/stop-writing-code-comments-28fef5272752

Reply:
I'm not sure which points of that lengthy list you're getting at. If you don't like the comment I can remove it. The code is not hard to understand and I can note down the reasoning why this is the desired behavior somewhat else.

Reply:
nevermind. I think we will talk about it another day :)